sdk/eventhub/eventhubs-checkpointstore-table/test/tables-checkpointstore.spec.ts

```diff
@@ -0,0 +1,556 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import chai from "chai";
+import * as dotenv from "dotenv";
+const should = chai.should();
+import { TableCheckpointStore } from "../src";
+import debugModule from "debug";
+const debug = debugModule("azure:event-hubs:partitionPump");
+import { Checkpoint, PartitionOwnership } from "@azure/event-hubs";
+import { EnvVarKeys, getEnvVars } from "./utils/testUtils";
+import { TableServiceClient, AzureNamedKeyCredential, TableClient } from "@azure/data-tables";
+import { CheckpointEntity, PartitionOwnershipEntity } from "../src/tableCheckpointStore";
+
+const env = getEnvVars();
+dotenv.config();
+
+/* test to show that test framework is set up well */
+describe("TableCheckpointStore", () => {
+  it("is exported from the package", () => {
+    should.exist(TableCheckpointStore, "Expected TableCheckpointStore to be exported.");
+  });
+});
+
+const service = {
+  storageConnectionString: env[EnvVarKeys.STORAGE_CONNECTION_STRING],
+  storageAccountName: env[EnvVarKeys.STORAGE_ACCOUNT_NAME],
+  storageAccountKey: env[EnvVarKeys.STORAGE_ACCOUNT_KEY],
+  tableName: env[EnvVarKeys.TABLENAME]
+};
+
+const credential = new AzureNamedKeyCredential(
+  service.storageAccountName,
+  service.storageAccountKey
+);
+const serviceClient = new TableServiceClient(
+  `https://${service.storageAccountName}.table.core.windows.net`,
+  credential
+);
+
+describe("TableCheckpointStore", function(): void {
+  let client: TableClient;
+  let table_name: string;
+
+  describe("Runs tests on table with no entities", function() {
+    table_name = `table${new Date().getTime()}`;
+    beforeEach("creating table", async () => {
+      await serviceClient.createTable(table_name);
+    });
+
+    client = new TableClient(
+      `https://${service.storageAccountName}.table.core.windows.net`,
+      table_name,
+      credential
+    );
+
+    afterEach(async () => {
+      await serviceClient.deleteTable(table_name);
+    });
+
+    describe("listOwnership", function() {
+      it("listOwnership should return an empty array", async function(): Promise<void> {
+        const checkpointStore = new TableCheckpointStore(client);
+        const listOwnership = await checkpointStore.listOwnership(
+          "test.servicebus.windows.net",
+          "testHub",
+          "testConsumerGroup"
+        );
+        should.equal(listOwnership.length, 0);
+      });
+    });
+
+    describe("updateOwnership", function() {
```

Comment:
This describe says `updateOwnership` but you're testing `updateCheckpoint`. 

I think you should have a test for each of the 4 public methods that ensures they work as expected, then you can add error cases as well (e.g. incorrect etag).

Reply:
does order in which i tested the methods matter ? Because I have tests showing that each one of them work fine.

Reply:
also thanks . it is supposed to be updateCheckpoint

Reply:
The order doesn't matter. Under the `Runs test on table with no entities` section, I only see the test here for `listOwnership` and `updateCheckpoint` with an error. I don't see `listCheckpoints`, `updateCheckpoint` successfully, or `claimOwnership`.

Reply:
i implemented updateCheckpoint where one does not exist, in another block .
same for claimownership.

for now I have added the test for listOwnership